x/qgb/keeper/keeper_data_commitment.go

```diff
@@ -52,3 +63,23 @@ func (k Keeper) GetDataCommitmentForHeight(ctx sdk.Context, height uint64) (type
 	}
 	return types.DataCommitment{}, fmt.Errorf("data commitment for height not found")
 }
+
+// GetLastDataCommitment returns the last data commitment.
+func (k Keeper) GetLastDataCommitment(ctx sdk.Context) (types.DataCommitment, error) {
+	latestNonce := k.GetLatestAttestationNonce(ctx)
+	for i := uint64(0); i < latestNonce; i++ {
```

Comment:
[no change needed] sorry for delayed review but this iteration may read easier with

```go
	for i := latestNonce; i >= 0; i-- {
```

because the contents of this for loop care about iterating from the latest nonce down to 0. This proposal removes the need for subtraction in the body of the for loop